### PR TITLE
Article on factorial of large numbers

### LINKED ISSUE #155: Doc for factorial of very large numbers under Competitive Programming
Would like to add doc for factorial of very large numbers under Competitive Programming.
Please assign this to me under GSSOC'21.
Thanks!

### PR DESCRIPTION
**Pull request by satcasm**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

Article on factorial of large numbers

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Fixes #155

### CODE REVIEW

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+#### Output:
+
+```
+Factorial of 51 is : 1551118753287382280224243016469303211063259720016986112000000000000
+```
+
+## Complexity Analysis
+
+- Time  complexity - O(n^2)
+- Space complexity - O(log(n!))
```
Comment (nidh-ish) [inline docs/competitive-programming/factorial-of-large-number.mdx:196]: Keep the values in the backticks.

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+	cin >> num;
+    largeFactorial(num);
+    return 0;
+}
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
```
Comment (nidh-ish) [inline docs/competitive-programming/factorial-of-large-number.mdx:132]: Make I/O the same as the given sample.

**Review by nidh-ish — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/factorial-of-large-number.mdx (hunk 2 above)
Comment (satcasm) [reply docs/competitive-programming/factorial-of-large-number.mdx:132]: Sure thing.

**Comment by satcasm** [comment]
@HarshCasper please review my PR.

**Review by nidh-ish — APPROVED** [review]
LGTM.

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+		Set product to res[i] * x + carry
+		Set res[i] to product % 10
+		Set carry to product/10
+	ENDFOR
+	WHILE carry exists
+		push carry%10 to res
+		divide carry by 10
+		increment rsize
+	ENDWHILE
+	RETURN rsize
```
Comment (HarshCasper) [inline docs/competitive-programming/factorial-of-large-number.mdx:56]: We don't need pseudo-code here. An actual text-based explanation would be appreciated.

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 4]
```diff
@@ -0,0 +1,201 @@
+---
+title: Factorial of very large number
```
Comment (HarshCasper) [inline docs/competitive-programming/factorial-of-large-number.mdx:2]: Keep it `Factorial of Large Number`

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 5]
```diff
@@ -0,0 +1,201 @@
+---
+title: Factorial of very large number 
+slug: /competitive-programming/factorial-of-large-number
```
Comment (HarshCasper) [inline docs/competitive-programming/factorial-of-large-number.mdx:3]: Update the slug accordingly

File: docs/competitive-programming/factorial-of-large-number.mdx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+## Explanation
+
+For a non-negative integer say `n`,the factorial of `n` is defined as follows:
+
+```
+Factorial(n) = n * (n - 1) * (n - 2) * . . . . . 3  * 2 * 1
+```
+
+Or
```
Comment (HarshCasper) [inline docs/competitive-programming/factorial-of-large-number.mdx:21]: Use `alternatively`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/competitive-programming/factorial-of-large-number.mdx (hunk 3 above)
Comment (satcasm) [reply docs/competitive-programming/factorial-of-large-number.mdx:56]: Sure thing.

**Comment by satcasm** [comment]
@HarshCasper made the desired changes have a look.
Thanks!

**Review by vybhav72954 — APPROVED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**